### DISCUSSION TITLE
Image transport for compressed video messages

### DISCUSSION THREAD

**Discussion by rcodddow** [foxglove/community#1021] [Q&A]
Currently I am not aware of any way to generate foxglove CompressedVideo messages or convert CompressedVideo messages to sensor_msgs/Image. Is there a compressvideo image transport plugin that you could make available ?

Im not against writing my own as the [blog post](https://foxglove.dev/blog/announcing-h264-support-in-foxglove) suggests forking the ffmpeg_image_transport package and making the necessary changes to the message structure. However, I would like to avoid this if possible.

**Comment by jtbandes** [comment]
I believe someone else in the community has already worked on this: https://github.com/angsa-robotics/ffmpeg_image_transport

**Comment by berndpfrommer** [comment]
I just finished a [package](https://github.com/ros-misc-utilities/foxglove_compressed_video_transport) that does that. Submitted to rosdistro for indexing.

**Reply by defunctzombie** [reply]
This is awesome! Any desire to collaborate on a blog post to do a mini tutorial on how to use the package?

**Reply by berndpfrommer** [reply]
Sure, we can do, but I'd like to see the package at least tested before announcing it to a wider audience. I haven't used foxglove yet so I don't know if it works as expected. We need to make sure the encoding parameters are set correctly.
